### DISCUSSION TITLE
Does golangci-lint make use of the go module cache?

### DISCUSSION THREAD

**Discussion by mpreu** [golangci/golangci-lint#3801] [Q&A] [answered]
Hi,

I am currently setting up a custom CI integration of `golangci-lint` using its container image. Here, I am wondering if `golangci-lint` needs access to the `go` module cache for performance (which I would have to provide externally in my use-case)?

I am asking, because the official GitHub Action talks a bit about caching/restoring build and module artifacts and I saw some other integrations providing an explicit module cache across pipelines. But I could not find anything in the documentation whether it is required or not. Is it enough to just provide a location for the `golangci-lint` specific cache to get max performance?

**Comment by utrack** [comment]
For the future readers - no it's not at the moment. It is set up via the envvar GOLANGCI_LINT_CACHE, which defaults to <.cache>/golangci-lint.

See the code here: https://github.com/golangci/golangci-lint/blob/master/internal/cache/default.go#L15

**Comment by ldez** [comment, marked as answer]
GitHub Action was handling the Go cache but since [v5.0.0](https://github.com/golangci/golangci-lint-action/releases/tag/v5.0.0) of `golangci-lint-action` this has been removed.

Golangci-lint by itself don't use directly the Go cache, it has its [own cache](https://github.com/golangci/golangci-lint-action/releases/tag/v5.0.0).
